pkg/cosign/kubernetes/apis/v1alpha1/clusterimagepolicy_types.go

```diff
@@ -0,0 +1,70 @@
+// +kubebuilder:validation:Optional
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/apis"
+)
+
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=clusterimagepolicies,scope=Cluster
+// +kubebuilder:storageversion
```

Comment:
I think we're pretty quickly get tired of juggling both of these, and I know when Knative Eventing briefly used controller-runtime they ended up having to maintain their own fork at times due to dependency hell.

Reply:
Sorry, I dont understand. Are you saying we should not use kubebuilder?

Reply:
I'm saying that if we use it, it shouldn't be "load bearing" and we should be prepared to rip it out if we need to (e.g. due to dep hell).

You can get basically every piece of codegen we need from K8s/Knative directly.

Reply:
@mattmoor Do you think we could use as example this https://github.com/knative-sandbox/sample-controller/blob/main/hack/update-codegen.sh#L34 and https://github.com/knative-sandbox/sample-controller/blob/main/pkg/apis/samples/v1alpha1/simple_deployment_types.go ? I think it does the same as the old kube-builder/controller-gen.

Reply:
That’s exactly what I’d use

Reply:
We started doing it using code-gen,  but had to vendor in  a lot of code. Is that acceptable to be pushed into cosign repo? There is no vendor folder as of now.

Reply:
Ah, I have tricks to avoid that needing an active vendor directory.

I have a couple free hours now, here's what I'll do to try and help bootstrap things here... 

I'll send a PR to `sigstore/cosign` that just sets up the layout of things for codegen with the appropriate layout and whatnot, and an ~empty Go type (enough to appease the compiler).

Then we can start to mix in the fields you have here, add the detailed comments from the schema, and sketch out defaulting and validation to match the semantics from the doc (with unit tests for different bits).

The vendoring bit is definitely a wrinkle I unfortunately don't have a good OSS example for, so I'll knock this off and tag y'all when I have something we can use as a starting point here.

Reply:
This has the basics, I'll fix codegen after I make dinner: https://github.com/sigstore/cosign/pull/1504 :)

Reply:
ok cool, we took a look into this. Wanted to know how the manifest for CRD is getting generated. I have used kubebuilder for this and I dont see how the manifest is generated in the `sample controller` repo. Can you point me to it please?

Reply:
cc @mattmoor  just tagging you as I dont know if you are still looking at this PR.

Reply:
so this: https://github.com/knative-sandbox/sample-controller/blob/6a2c2d197b4f37e6020ea19639a1077b7f6e50ab/config/300-simpledeployment.yaml is handwritten?

Reply:
oh it is okay, we can add it. We were looking to see examples of how it is generated, thats all.

Reply:
We can probably use `yq` to get this added to `./hack/update-codegen.sh`